File: sample/src/main/java/ru/bartwell/exfilepickersample/MainActivity.java
package ru.bartwell.exfilepickersample;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.AppCompatCheckBox;
import android.view.View;
import android.widget.RadioGroup;
import android.widget.TextView;

import ru.bartwell.exfilepicker.ExFilePicker;
import ru.bartwell.exfilepicker.data.ExFilePickerResult;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private static final int EX_FILE_PICKER_RESULT = 0;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.choose_button).setOnClickListener(this);
    }

    @Override
    public void onClick(@NonNull View v) {
        ExFilePicker exFilePicker = new ExFilePicker();
        if (((AppCompatCheckBox) findViewById(R.id.only_one_item)).isChecked()) {
            exFilePicker.setCanChooseOnlyOneItem(true);
        }
        if (((AppCompatCheckBox) findViewById(R.id.filter_listed)).isChecked()) {
            exFilePicker.setShowOnlyExtensions("jpg", "jpeg");
        }
        if (((AppCompatCheckBox) findViewById(R.id.filter_exclude)).isChecked()) {
            exFilePicker.setExceptExtensions("jpg");
        }
        if (((AppCompatCheckBox) findViewById(R.id.disable_new_folder_button)).isChecked()) {
            exFilePicker.setNewFolderButtonDisabled(true);
        }
        if (((AppCompatCheckBox) findViewById(R.id.disable_sort_button)).isChecked()) {
            exFilePicker.setSortButtonDisabled(true);
        }
        if (((AppCompatCheckBox) findViewById(R.id.enable_quit_button)).isChecked()) {
            exFilePicker.setQuitButtonEnabled(true);
        }
        if (((AppCompatCheckBox) findViewById(R.id.reverse_sorting)).isChecked()) {
            exFilePicker.setSortingType(ExFilePicker.SortingType.NAME_DESC);
        }
        if (((AppCompatCheckBox) findViewById(R.id.start_from_root)).isChecked()) {
            exFilePicker.setStartDirectory("/");
        }
        if (((AppCompatCheckBox) findViewById(R.id.first_item_as_up)).isChecked()) {
            exFilePicker.setUseFirstItemAsUpEnabled(true);
        }
        if (((AppCompatCheckBox) findViewById(R.id.hide_hidden_files)).isChecked()) {
            exFilePicker.setHideHiddenFilesEnabled(true);
        }
        int checkedChoiceRadio = ((RadioGroup) findViewById(R.id.choice_type)).getCheckedRadioButtonId();
        if (checkedChoiceRadio == R.id.choice_type_files) {
            exFilePicker.setChoiceType(ExFilePicker.ChoiceType.FILES);
        } else if (checkedChoiceRadio == R.id.choice_type_directories) {
            exFilePicker.setChoiceType(ExFilePicker.ChoiceType.DIRECTORIES);
        }
        exFilePicker.start(this, EX_FILE_PICKER_RESULT);
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        if (requestCode == EX_FILE_PICKER_RESULT) {
            ExFilePickerResult result = ExFilePickerResult.getFromIntent(data);
            if (result != null && result.getCount() > 0) {
                StringBuilder stringBuilder = new StringBuilder();
                for (int i = 0; i < result.getCount(); i++) {
                    stringBuilder.append(result.getNames().get(i));
                    if (i < result.getCount() - 1) stringBuilder.append(", ");
                }
                ((TextView) findViewById(R.id.result)).setText("Count: " + result.getCount() + "\n" + "Path: " + result.getPath() + "\n" + "Selected: " + stringBuilder.toString());
            }
        }
    }
}

File: sample/src/main/java/ru/bartwell/exfilepickersample/MainActivity.kt
package ru.bartwell.exfilepickersample

import android.os.Bundle
import android.widget.Toast
import androidx.activity.ComponentActivity
import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.compose.setContent
import androidx.activity.result.contract.ActivityResultContracts
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Button
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import ru.bartwell.exfilepicker.ExFilePicker
import ru.bartwell.exfilepicker.data.config.*
import ru.bartwell.exfilepickersample.items.CheckBoxItem
import ru.bartwell.exfilepickersample.items.RadioButtonItem
import ru.bartwell.exfilepickersample.ui.theme.ExFilePickerTheme

class MainActivity : ComponentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            ExFilePickerTheme {
                Surface(modifier = Modifier.fillMaxSize()) {
                    ScreenContent()
                }
            }
        }
    }
}

@Composable
private fun ScreenContent() {
    val radioOptions = listOf(
        stringResource(R.string.choice_type_all),
        stringResource(R.string.choice_type_files),
        stringResource(R.string.choice_type_directories)
    )
    val scrollState = rememberScrollState()
    val onlyOneCheckboxState = remember { mutableStateOf(false) }
    val filterListedCheckboxState = remember { mutableStateOf(false) }
    val filterExcludeCheckboxState = remember { mutableStateOf(false) }
    val newFolderCheckboxState = remember { mutableStateOf(false) }
    val disableSortCheckboxState = remember { mutableStateOf(false) }
    val enableQuitCheckboxState = remember { mutableStateOf(false) }
    val reverseSortingCheckboxState = remember { mutableStateOf(false) }
    val startFromRootCheckboxState = remember { mutableStateOf(false) }
    val firstItemAsUpCheckboxState = remember { mutableStateOf(false) }
    val hideHiddenFilesCheckboxState = remember { mutableStateOf(false) }
    val (selectedOption, onOptionSelected) = remember { mutableStateOf(radioOptions[0]) }
    val showPickerState = remember { mutableStateOf(false) }

    val context = LocalContext.current
    val exFilePicker = ExFilePicker()
    val launcher = rememberLauncherForActivityResult(ActivityResultContracts.StartActivityForResult()) {
        showPickerState.value = false
        val result = exFilePicker.getResult(it.data)?.toList().toString()
        Toast.makeText(context, "Result = $result", Toast.LENGTH_SHORT).show()
    }
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .verticalScroll(scrollState),
    ) {
        CheckBoxItem(
            title = stringResource(R.string.only_one_item), onCheckedChange = { onlyOneCheckboxState.value = it },
            isChecked = onlyOneCheckboxState.value
        )
        CheckBoxItem(
            title = stringResource(R.string.filter_listed), onCheckedChange = { filterListedCheckboxState.value = it },
            isChecked = filterListedCheckboxState.value
        )
        CheckBoxItem(
            title = stringResource(R.string.filter_exclude), onCheckedChange = { filterExcludeCheckboxState.value = it },
            isChecked = filterExcludeCheckboxState.value
        )
        CheckBoxItem(
            title = stringResource(R.string.disable_new_folder_button), onCheckedChange = { newFolderCheckboxState.value = it },
            isChecked = newFolderCheckboxState.value
        )
        CheckBoxItem(
            title = stringResource(R.string.disable_sort_button), onCheckedChange = { disableSortCheckboxState.value = it },
            isChecked = disableSortCheckboxState.value
        )
        CheckBoxItem(
            title = stringResource(R.string.enable_quit_button), onCheckedChange = { enableQuitCheckboxState.value = it },
            isChecked = enableQuitCheckboxState.value
        )
        CheckBoxItem(
            title = stringResource(R.string.reverse_sorting), onCheckedChange = { reverseSortingCheckboxState.value = it },
            isChecked = reverseSortingCheckboxState.value
        )
        CheckBoxItem(
            title = stringResource(R.string.start_from_root), onCheckedChange = { startFromRootCheckboxState.value = it },
            isChecked = startFromRootCheckboxState.value
        )
        CheckBoxItem(
            title = stringResource(R.string.first_item_as_up), onCheckedChange = { firstItemAsUpCheckboxState.value = it },
            isChecked = firstItemAsUpCheckboxState.value
        )
        CheckBoxItem(
            title = stringResource(R.string.hide_hidden_files), onCheckedChange = { hideHiddenFilesCheckboxState.value = it },
            isChecked = hideHiddenFilesCheckboxState.value
        )

        Text(
            text = stringResource(R.string.choice_type),
            modifier = Modifier.padding(start = 20.dp, top = 0.dp, end = 0.dp, bottom = 20.dp)
        )
        radioOptions.forEach { text ->
            RadioButtonItem(
                selected = (text == selectedOption),
                title = text,
                onSelectedChange = {
                    onOptionSelected(text)
                }
            )
        }

        Box(
            modifier = Modifier
                .fillMaxSize()
                .padding(horizontal = 20.dp, vertical = 20.dp)
        ) {
            Button(onClick = { showPickerState.value = true }) {
                Text(text = stringResource(R.string.choose_files))
            }
            if (showPickerState.value) {
                val onlyExtensions = if (filterListedCheckboxState.value) {
                    arrayOf("*.jpeg", "*.jpg")
                } else {
                    emptyArray()
                }
                val exceptExtensions = if (filterExcludeCheckboxState.value) {
                    arrayOf("*.jpg")
                } else {
                    emptyArray()
                }
                val sortingType = if (reverseSortingCheckboxState.value) {
                    SortingType.NAME_DESC
                } else {
                    SortingType.NAME_ASC
                }
                val startDirectory = if (startFromRootCheckboxState.value) {
                    "/"
                } else {
                    null
                }
                val choiceType = when (selectedOption) {
                    stringResource(R.string.choice_type_all) -> {
                        ChoiceType.ALL
                    }
                    stringResource(R.string.choice_type_files) -> {
                        ChoiceType.FILES
                    }
                    else -> {
                        ChoiceType.DIRECTORIES
                    }
                }
                val config = Config(
                    canChooseOnlyOneItem = onlyOneCheckboxState.value,
                    showOnlyExtensions = onlyExtensions,
                    exceptExtensions = exceptExtensions,
                    isNewFolderButtonDisabled = newFolderCheckboxState.value,
                    isSortButtonDisabled = disableSortCheckboxState.value,
                    isQuitButtonEnabled = enableQuitCheckboxState.value,
                    sortingType = sortingType,
                    startDirectory = startDirectory,
                    useFirstItemAsUpEnabled = firstItemAsUpCheckboxState.value,
                    hideHiddenFilesEnabled = hideHiddenFilesCheckboxState.value,
                    choiceType = choiceType,
                    uiConfig = UiConfig(
                        darkMode = DarkMode.DARK
                    )
                )
                val intent = exFilePicker.createIntent(context, config)
                launcher.launch(intent)
            }
        }
    }
}

@Preview(showBackground = true)
@Composable
private fun Preview() {
    ExFilePickerTheme {
        ScreenContent()
    }
}
